vllm/v1/core/sched/scheduler.py

```diff
@@ -150,7 +156,8 @@ def schedule(self) -> SchedulerOutput:
                 req_index += 1
                 continue
 
-            num_new_tokens = (request.num_tokens_with_spec -
+            num_new_tokens = (request.num_tokens_with_spec +
+                              self.num_lookahead_slots -
```

Comment:
IIUC, this is not correct because num_lookahead_slots is only for memory allocation, not about how many tokens to be batched in this step.

Reply:
Can I ask how is `self.num_lookahead_slots` different from `len(request.spec_token_ids)`? 
Isn't `request.num_tokens_with_spec` is taking into account the number of draft token for the req as it is def as `len(request._all_token_ids) + len(request.spec_token_ids)`?

Reply:
Ah, probably `self.num_lookahead_slots` is needed for kv cache of transformer block in eagle proposer? If so, then we need a bigger kvc for eagle module when doing prefill since the entire prompt is processed by the eagle module.